### PR TITLE
Convert DefaultHeaders plugin to new API

### PR DESCRIPTION
**Pull request by hhariri**
Convert DefaultHeaders plugin to new API

### CODE REVIEW

**Review by e5l — APPROVED** [review]
lgtm

**Comment by e5l** [comment]
Please rebase on the main branch and run `./gradlew apiDump` with committing changes before merging

File: ktor-server/ktor-server-plugins/ktor-server-default-headers/jvm/src/io/ktor/server/plugins/DefaultHeaders.kt [hunk 1]
```diff
@@ -7,93 +7,72 @@ package io.ktor.server.plugins
...
-        public var clock: () -> Long = { System.currentTimeMillis() }
+/**
+ * Adds the standard `Date` and `Server` HTTP headers and provides the ability
+ * to add additional default headers into each response.
+ */
+public val DefaultHeaders: RouteScopedPlugin<DefaultHeadersConfig, PluginInstance> = createRouteScopedPlugin(
+    "DefaultHeaders",
+    createConfiguration = {
+        DefaultHeadersConfig()
+    }
```
Comment (Ololoshechkin) [inline ktor-server/ktor-server-plugins/ktor-server-default-headers/jvm/src/io/ktor/server/plugins/DefaultHeaders.kt:44]: Nit: I would just simply write `:: DefaultHeadersConfig`

**Review by Ololoshechkin — APPROVED** [review]

File: ktor-server/ktor-server-plugins/ktor-server-default-headers/jvm/src/io/ktor/server/plugins/DefaultHeaders.kt (hunk 1 above)
Comment (hhariri) [reply ktor-server/ktor-server-plugins/ktor-server-default-headers/jvm/src/io/ktor/server/plugins/DefaultHeaders.kt:44]: Actually the docs in the issue was using this style...I can change.

**Pull request merged**